detox/src/utils/ConfigurationResolver.js

```diff
@@ -0,0 +1,39 @@
+const path = require('path');
+const fs = require('fs');
+const packageJson = 'package.json';
+const detoxRc = '.detoxrc';
+
+class ConfigurationResolver {
+  constructor(cwd = process.cwd()) {
+    this.cwd = cwd;
+  }
+
+  getDetoxConfiguration(configPath) {
+    if (configPath) {
+      return this.loadConfiguration(configPath);
+    }
+
+    const { detox } = this.loadConfiguration(packageJson);
+    if (detox) {
+      return detox;
+    }
+    return this.loadDetoxrcConfiguration()
+  }
+
+  loadDetoxrcConfiguration() {
+    var data = fs.readFileSync(this.resolvePath(detoxRc)).toString();
```

Comment:
Is there a specific consideration of why you want to limit `.detoxrc` to JSON format only?
Personally, I'd love to be able to evaluate it as JS, I'm confident that some people would find how to leverage that for their purposes.

Reply:
So what do you suggest ? this can (and should) be separated and added in a later PR IMO.

Reply:
@rotemmiz , I suggest using plain `require()` instead on `JSON.parse(fs.readFileSync...`